fiftyone/utils/hf_hub.py

```diff
@@ -0,0 +1,1109 @@
+import fiftyone.core.dataset as fod
+import fiftyone.core.labels as fol
+import fiftyone.core.metadata as fom
+from fiftyone.core.sample import Sample
+import fiftyone.core.utils as fou
+import fiftyone.types as fot
+
+DATASETS_SERVER_URL = "https://datasets-server.huggingface.co"
+DEFAULT_MEDIA_TYPE = "image"
+DATASET_METADATA_FILENAMES = ("fiftyone.yml", "fiftyone.yaml")
+DATASETS_MAX_BATCH_SIZE = 100
+DEFAULT_IMAGE_FILEPATH_FEATURE = "image"
+FIFTYONE_BUILTIN_FIELDS = ("id", "filepath", "tags", "metadata")
+REPO_TYPE = "dataset"
+SUPPORTED_DTYPES = (
+    "int8",
+    "int16",
+    "int32",
+    "int64",
+    "float16",
+    "float32",
+    "float64",
+    "bool",
+    "string",
+)
+
+logger = logging.getLogger(__name__)
+
+
+def get_cache():
+    g = globals()
+    if "_files_to_download" not in g:
+        g["_files_to_download"] = []
+
+    return g["_files_to_download"]
+
+
+def push_to_hub(
+    dataset,
+    repo_name,
+    description=None,
+    license=None,
+    tags=None,
+    private=True,
+    exist_ok=False,
+    dataset_type=None,
+    label_field=None,
+    frame_labels_field=None,
+    **data_card_kwargs,
+):
+    """Push a FiftyOne dataset to the Hugging Face Hub.
+
+    Args:
+        dataset: a FiftyOne dataset
+        repo_name: the name of the dataset repo to create
+        description (None): a description of the dataset
+        license (None): the license of the dataset
+        tags (None): a list of tags for the dataset
+        private (True): whether the repo should be private
+        exist_ok (False): if True, do not raise an error if repo already exists.
+        dataset_type (None): the type of the dataset to create
+        label_field (None): controls the label field(s) to export. Only
+            applicable to labeled datasets. Can be any of the following:
+            - the name of a label field to export
+            - a glob pattern of label field(s) to export
+            - a list or tuple of label field(s) to export
+            - a dictionary mapping label field names to keys to use when
+              constructing the label dictionaries to pass to the exporter
+        frame_labels_field (None): controls the frame label field(s) to export.
+            The "frames." prefix is optional. Only applicable to labeled video
+            datasets. Can be any of the following:
+            - the name of a frame label field to export
+            - a glob pattern of frame label field(s) to export
+            - a list or tuple of frame label field(s) to export
+            - a dictionary mapping frame label field names to keys to use when
+              constructing the frame label dictionaries to pass to the exporter
+        data_card_kwargs: additional keyword arguments to pass to the
+            `DatasetCard` constructor
+    """
+    if dataset_type is None:
+        dataset_type = fot.FiftyOneDataset
+
+    ### export the dataset to a temp local dir
+    tmp_dir = f"/tmp/{repo_name}"
```

Comment:
We have a `TempDir` context manager for cases like this:

```py
import eta.core.utils as etau

with etau.TempDir() as tmp_dir:
    dataset.export(...)

    # other stuff

# tmp_dir is automatically deleted when the context exits
```

The context manager ensures that things will always get cleaned up, even if there's an error or the user cancels the operation. And it handles macOS vs Windows path considerations for you.